lib/iris/util.py

```diff
@@ -406,17 +406,24 @@ def array_equal(array1, array2, withnans=False):
 
     def normalise_array(array):
         if not is_lazy_data(array):
-            array = np.asarray(array)
+            if not ma.isMaskedArray(array):
```

Comment:
I think that the behaviour changes to this function can reasonably be considered as "fixes" and not "breaking change".

But, can you please add something to the docstring to make the operation on masked arrays explicit ?
The existing docstring simply doesn't cover it, and suggests it behaves like `np.array_equal`, which is not really true any more, at least in this respect :
```
>>> np.array_equal([5, 6], np.ma.masked_array([5, 6], mask=[0,1]))
True
```

So, I think the whats-new comment is totally clear about the masked behaviour, but the method docstring is not (!)